src/main/java/bio/terra/workspace/common/model/WorkspaceStage.java

```diff
@@ -0,0 +1,38 @@
+package bio.terra.workspace.common.model;
+
+import bio.terra.workspace.generated.model.WorkspaceStageModel;
+import com.google.common.collect.BiMap;
+import com.google.common.collect.EnumBiMap;
+
+/**
+ * Internal representation of the workspace stage used for gating MC Terra features from Rawls
+ * workspaces.
+ */
+public enum WorkspaceStage {
+  RAWLS_WORKSPACE,
+  MC_WORKSPACE;
+
+  private static final BiMap<WorkspaceStage, WorkspaceStageModel> stageMap =
+      EnumBiMap.create(WorkspaceStage.class, WorkspaceStageModel.class);
+
+  static {
+    stageMap.put(WorkspaceStage.RAWLS_WORKSPACE, WorkspaceStageModel.RAWLS_WORKSPACE);
+    stageMap.put(WorkspaceStage.MC_WORKSPACE, WorkspaceStageModel.MC_WORKSPACE);
+  }
+
+  public static WorkspaceStage fromApiModel(WorkspaceStageModel modelEnum) {
+    WorkspaceStage stage = stageMap.inverse().get(modelEnum);
+    if (modelEnum == null) {
+      return WorkspaceStage.RAWLS_WORKSPACE;
+    }
+    return stage;
+  }
+
+  public WorkspaceStageModel toApiModel() {
+    WorkspaceStageModel stage = stageMap.get(this);
+    if (stage == null) {
+      return WorkspaceStageModel.RAWLS_WORKSPACE;
+    }
```

Comment:
why do we need default values in the conversion?
If it's to support rows that already exist in the db, better to take the one time hit of doing a db migration, or handle the backwards compatibility null conversion in the DAO. If it's to support request serialization, I think we can specify open api default values. (Though, even then, we would only need the default from the model). If we never expect the arguments to be null, this ends up being confusing. If we add new values but forget to update this mapping, we'll silently use the defaults. Loudly failing early makes it more likely for us to find a bug before it's an issue.
It's much easier to reason about a non-null world, but it takes more effort up front to get there.

Reply:
I wanted requests with no value specified to default to RAWLS_WORKSPACE so that existing clients didn't need to know about the change. It's not a large change to update the client library Rawls is using, but it felt like Rawls shouldn't need to know about this toggle for features it isn't using. The default you specify in openAPI for request body schemas only affects the UI, an unspecified value still comes in as null (I think it's different for path/query params, but I'm not sure).

That said postgres supports default values for columns, so we could translate a null modelEnum to a null WorkspaceStageModel and when we write it to the DB we'd have RAWLS_WORKSPACE and wouldn't need the internal model -> api model default. I didn't like the idea of passing null around like that, but it would give a clearer warning in the case where we update the api enum without changing the model enum. I'm fine changing that if you think the warning is worth it

Reply:
Not needing to update the Rawls client sounds reasonable. Sorry, I was wrong about the defaults.
If the goal is to support backwards compatibility with existing client libraries, the right place to handle the unspecified null value is in the API layer where the request is received. We're trying to map existing create workspace requests -> RAWLS_WORKSPACE requests. That's a concern about the representation in the create workspace API and how we can change it over time. It's not a general conversion principle, or a stance that any default null value should be treated as RAWLS_WORKSPACE. If we added another endpoint that fetched all my workspaces in a separate stage, we wouldn't want 'null' to mean RAWLS_WORKSPACE. So this null/unspecified handling code should live in the controller that handles the API, and be used only in the endpoint we're concerned about.
This is a good, albeit minor, example of the benefits of having distinct layers. With a conversion in the API layer, none of the rest of the service needs to worry about the fact that the create workspace request has this support. When I'm reading the rest of the service, I don't need to worry about, oh, maybe this value was null and it got converted.

That sounds like we only need the default conversion for model -> internal.
I think the internal -> model default conversion is for the rows already in the DB. For that, I still think it's simpler to write liquibase to update all existing rows to RAWLS_WORKSPACE, as if they had all been created after this code change. Using postgres to interpret default null values does sound more confusing.

Reply:
Sounds good, moved the default conversion into the controller endpoint and removed the internal->model conversion altogether. I also switched the liquibase to update all existing rows to RAWLS_WORKSPACE and not do anything special when workspace stage isn't specified.